src/__smoke-test__/smoke-test.spec.ts

```diff
@@ -0,0 +1,446 @@
+import { test, expect } from '@playwright/test';
+import {
+    HTTP_EVENT_TYPE,
+    JS_ERROR_EVENT_TYPE,
+    LCP_EVENT_TYPE,
+    FID_EVENT_TYPE,
+    PERFORMANCE_NAVIGATION_EVENT_TYPE,
+    PERFORMANCE_RESOURCE_EVENT_TYPE,
+    PAGE_VIEW_EVENT_TYPE,
+    SESSION_START_EVENT_TYPE
+} from '../plugins/utils/constant';
+
+// Secrete constant values retrieved from GitHub
+const endpoint = '${process.env.ENDPOINT}';
+const monitorId = '${process.env.MONITOR_ID}';
+const url = '${process.env.SMOKE_URL}';
+
+function parseBatches(requestData) {
+    const output = [];
+    for (let i = 0; i < requestData.length; i++) {
+        const parsed = JSON.parse(requestData[i]).RumEvents;
+        for (let j = 0; j < parsed.length; j++) {
+            output.push(parsed[j]);
+        }
+    }
+    return output;
+}
+
+function addEventsToList(parsedPayload, eventList) {
+    for (let i = 0; i < parsedPayload.length; i++) {
+        eventList.push(parsedPayload[i].id);
+    }
+}
+
+function getEventsByType(requestBody, eventType) {
+    return requestBody.filter((e) => e.type === eventType);
+}
+
+function filterSuccessfulResponse(response, requestBody) {
+    // filter out irrelevant, unsuccessful network request/response
+    const request = response.request();
+    if (
+        request &&
+        response.status() === 200 &&
+        request.method() === 'POST' &&
+        response.url() === endpoint + monitorId + '/' &&
+        request.postData()
+    ) {
+        requestBody.push(request.postData());
+    }
+}
+
+// Timestamp to filter events fetched via getAppMonitorData API
+const time = Date.now();
+
+test.describe('Smoke test fixtures', () => {
+    let page;
+    const eventIds = [];
+
+    // Use single browser context to prevent multiple sessions per test
+    test.beforeAll(async ({ browser }) => {
+        const context = await browser.newContext();
+        page = await context.newPage();
+        await page.goto(url);
```

Comment:
issue: Tests should be independent of each other

Reply:
I agree that tests should be independent, but there are three reasons as to why I wanted to have the test cases use the same browser session:
1. unique browser session leads to browser opening and closing for all the test cases
2. certain events such as session_start_event will be emitted over and over
3. when we perform the tests manually, we do not open and close the page after clicking/testing one button; instead, we go through all the buttons under one session